packages/grafana-runtime/src/analytics/plugins/eventProperties.ts

```diff
@@ -0,0 +1,35 @@
+import { DataSourceApi, PluginMeta, PluginType } from '@grafana/data';
+
+import { config } from '../../config';
+
+export type PluginEventProperties = {
+  grafana_version: string;
+  plugin_type: PluginType;
+  plugin_version: string;
+  plugin_id: string;
+  plugin_name: string;
+};
+
+export function createPluginEventProperties(meta: PluginMeta): PluginEventProperties {
+  return {
+    grafana_version: config.buildInfo.version,
+    plugin_type: meta.type,
+    plugin_version: meta.info.version,
+    plugin_id: meta.id,
+    plugin_name: meta.name,
```

Comment:
I like that idea, can you expand on this idea a bit more

Reply:
For datasources for instance, we could group results by category to answer questions like are `tsdb` / `cloud` / `sql` type datasources easier to use / most used / trigger the most errors... depending on the interactions we track.

It's hard to find an exact use case where it would be something that we would want to look at though, so I would not mind adding it at a later stage when the actual need arises 😉

Reply:
I like the idea but not sure we have that kind of category for each data source

Reply:
I see that we have that for the internal plugins but it is rarely specified for external plugins.